mypy/checkmember.py

```diff
@@ -109,10 +109,17 @@ def analyze_member_access(name: str,
     elif isinstance(typ, NoneTyp):
         if chk.should_suppress_optional_error([typ]):
             return AnyType(TypeOfAny.from_error)
-        # The only attribute NoneType has are those it inherits from object
-        return analyze_member_access(name, builtin_type('builtins.object'), node, is_lvalue,
-                                     is_super, is_operator, builtin_type, not_ready_callback, msg,
-                                     original_type=original_type, chk=chk)
+        if name == '__bool__':
```

Comment:
Note that that this special case should be added only on Python 3.

Please also add a test that `None.__bool__` fails on Python 2.

Reply:
> Note that that this special case should be added only on Python 3.

Oops, I'd missed that. Thanks, fixed.

> Please also add a test that `None.__bool__` fails on Python 2.

Done.